orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -685,6 +685,9 @@ def _init_session(self):
         if self.options.utils.run_symbolic_shape_infer:
             self._onnx_model = SymbolicShapeInference.infer_shapes(self._onnx_model, auto_merge=True, guess_output_rank=True)
 
+        # old ort session may already exists and occupies GPU memory when creating new session, this may cause OOM error.
```

Comment:
Great finding. 
NIP, shall this code be at where self._training_session is assigned?:
self._training_session = ort.TrainingSession(self._onnx_model.SerializeToString(),

Reply:
Agreed

Reply:
good point!  just resolved it.